graphql_compiler/scripts/generate_test_sql/animals.py

```diff
@@ -0,0 +1,82 @@
+import random
+
+from .species import SPECIES_LIST
+from .utils import (create_edge_statement, create_name, create_vertex_statement, get_uuid,
+                    strip_index_from_name)
+
+
+NUM_INITIAL_ANIMALS = 5
+NUM_GENERATIONS = 10
+NUM_PARENTS = 3
+
+
+def _create_animal_statement(animal_name):
+    """Return a SQL statement to create an animal vertex."""
+    fields_dict = {'name': animal_name, 'uuid': get_uuid()}
+    return create_vertex_statement('Animal', fields_dict)
+
+
+def _create_animal_parent_of_statement(from_name, to_name):
+    """Return a SQL statement to create an Animal_ParentOf edge."""
+    return create_edge_statement('Animal_ParentOf', 'Animal', from_name, 'Animal', to_name)
+
+
+def _create_animal_of_species_statement(from_name, to_name):
+    """Return a SQL statement to create an Animal_OfSpecies edge."""
+    return create_edge_statement('Animal_OfSpecies', 'Animal', from_name, 'Species', to_name)
+
+
+def _get_initial_animal_generators(species_name, current_animal_names):
+    """Return a list of SQL statements to create initial animals for a given species."""
+    generator_list = []
+    for index in range(NUM_INITIAL_ANIMALS):
+        animal_name = create_name(species_name, str(index))
+        current_animal_names.append(animal_name)
+        generator_list.append(_create_animal_statement(animal_name))
+    return generator_list
+
+
+def _get_new_parents(current_animal_names, previous_parent_sets, num_parents):
+    """Return a list of `num_parents` parent names that is not present in `previous_parent_sets`."""
+    while True:
+        new_parent_names = frozenset(random.sample(current_animal_names, num_parents))
+        # Duplicating a set of parents could result in Animals with the same names.
+        # This would invalidate unique selection of Animals by name.
+        if new_parent_names not in previous_parent_sets:
+            return new_parent_names
+
+
+def get_animal_generators():
+    """Return a list of SQL statements to create animal vertices and their corresponding edges."""
+    generator_list = []
+    species_to_names = {}
+    previous_parent_sets = set()
+
+    for species_name in SPECIES_LIST:
+        current_animal_names = []
+        species_to_names[species_name] = current_animal_names
+        generator_list.extend(_get_initial_animal_generators(species_name, current_animal_names))
+
+        for _ in range(NUM_GENERATIONS):
+            new_parent_names = _get_new_parents(
+                current_animal_names, previous_parent_sets, NUM_PARENTS)
+            previous_parent_sets.add(new_parent_names)
+
+            parent_indices = [
+                index
+                for _, index in [
+                    strip_index_from_name(parent_name) for parent_name in new_parent_names
```

Comment:
always put the `for` on a new line, unless it's an extremely short list comprehension